### PR TITLE
Convert OSM pbf file to osm rasters and roads csv

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[OSM pipeline](https://trello.com/c/ka7OwdSB/29-osm-pipeline)

File: README.md [hunk 1]
```diff
@@ -22,20 +26,23 @@ OSM_DATA_SOURCE=https://osm.openarchive.site/planet-latest.osm.pbf
...
+               [--extent EXTENT] [--backup_step_data]
 
 optional arguments:
   -h, --help            show this help message and exit
   -d TASKDATE, --taskdate TASKDATE
   -f OSM_FILE, --osm_file OSM_FILE
-                        Add local path to OSM source file. If not provided, file will be downloaded
+                        Add local path to OSM source file. If not provided,
+                        file will be downloaded
   -u OSM_URL, --osm_url OSM_URL
```
Comment (saanobhaai) [inline README.md:39]: Should put `default` param on all args that then get documented here

File: src/config.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+road_tags = set((
```
Comment (saanobhaai) [inline src/config.py]: Ideally this set of tags would only be specified once (i.e. combined with the set in the osmium config)

File: README.md [hunk 3]
```diff
@@ -3,11 +3,15 @@ HII OSM CSV
...
 
-1. Fetch Open Street Map PBF file (default: https://osm.openarchive.site/planet-latest.osm.pbf)
-2. Convert PBF file to CSV file, filtered by OSM tags (see: `config.py`)
-3. Upload filtered CSV file to Google Cloud Storage
-4. Using Earth Engine CLI load CSV as table in Earth Engine
-
+1. Fetch OSM pbf file
+2. Convert PBF file -> Text file (filter by attribute/tag list)
+3.
+  a)Split up text file by attribute and tags and 1 million row CSV files
```
Comment (saanobhaai) [inline README.md]: a) Split up text file into one CSV file per attribute/tag combination per 1 million rows

File: README.md [hunk 4]
```diff
@@ -3,11 +3,15 @@ HII OSM CSV
...
+  b)Clean geometry and write out road tags to a roads CSV file
+4. Rasterize each CSV file.
+5. Merge all tiff into 1 multiband tiff file
```
Comment (saanobhaai) [inline README.md]: well, it's 1 tiff split into global slices

File: src/task.py [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+
+    return run
+
+
+def _rasterize(
+    in_file: Union[str, Path],
+    output_path: Union[str, Path],
+    output_bounds: Optional[List[float]] = None,
+) -> Path:
+    output_bounds = output_bounds or [-180.0, -90.0, 180.0, 90.0]
```
Comment (saanobhaai) [inline src/task.py]: Is this why output footprints are the way we're seeing them in ee?

File: src/task.py [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+    _asset_prefix = f"projects/{HIITask.ee_project}/{ee_osm_root}"
+
+    MIN_GEOM_AREA = 5  # in meters
+    POLYGON_PRECISION = 5
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(self, *args, **kwargs)
+
+        self._args = kwargs
+        self.max_rows = 1000000
```
Comment (saanobhaai) [inline src/task.py]: Extract out of `__init__` as constant?

File: src/task.py [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+
+        _extent = self._args.get("extent")
+        if _extent:
```
Comment (saanobhaai) [inline src/task.py:106]: Or maybe this test isn't behaving as expected? i.e. `argparse` is returning something truthy?

File: src/task.py [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+    )
+
+    parser.add_argument(
+        "-w",
+        "--working_dir",
+        type=str,
+        help="Working directory to store files and directories during processing.",
+    )
+
+    parser.add_argument("--extent", type=str, help="Output geographic bounds.")
```
Comment (saanobhaai) [inline src/task.py]: See earlier comments about defaults and also extent...

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Just some minor comments...

File: src/osmium_config.json [hunk 9]
```diff
@@ -145,5 +145,36 @@
         "railway=station",
         "railway=subway",
         "railway=tram"
-    ]
+    ],
+    "road_tags": {
+        "highway=bridleway": ["highway", "ridleway"],
```
Comment (saanobhaai) [inline src/osmium_config.json]: Just noticed the tag typo... wonder if there's any way to check them all without doing it manually...

File: src/task.py [hunk 10]
```diff
@@ -457,27 +465,36 @@ def calc(self):
...
         type=str,
+        default="/tmp",
         help="Working directory to store files and directories during processing.",
     )
 
-    parser.add_argument("--extent", type=str, help="Output geographic bounds.")
+    parser.add_argument(
+        "--extent",
+        type=str,
+        default="-180.0,-58.0,180.0,84.0",
```
Comment (saanobhaai) [inline src/task.py]: Couldn't this be `str(HIITask.extent)`?

**Review by saanobhaai — CHANGES_REQUESTED** [review]
saw a typo, and still wondering about extent...

**Pull request merged**